crates/viewer/re_ui/src/notifications.rs

```diff
@@ -0,0 +1,393 @@
+    }
+
+    pub fn add(&mut self, level: Level, text: impl Into<String>) {
+        if !is_relevant(level) {
+            return;
+        }
+
+        self.data.push(Notification {
+            level: level.into(),
+            text: text.into(),
+
+            created_at: Zoned::now(),
+            ttl: base_ttl(),
+        });
+        self.has_unread_notifications = true;
+    }
+
+    pub fn success(&mut self, text: impl Into<String>) {
+        self.data.push(Notification {
+            level: NotificationLevel::Success,
+            text: text.into(),
+
+            created_at: Zoned::now(),
+            ttl: base_ttl(),
+        });
+    }
+
+    pub fn ui(&mut self, egui_ctx: &egui::Context, is_panel_visible: &mut bool) {
+        if *is_panel_visible {
+            let escape_pressed =
+                egui_ctx.input_mut(|i| i.consume_key(egui::Modifiers::NONE, egui::Key::Escape));
+            if escape_pressed {
+                *is_panel_visible = false;
+            }
+
+            self.has_unread_notifications = false;
+        }
+
+        self.panel.show(egui_ctx, &mut self.data, is_panel_visible);
+        self.toasts.show(egui_ctx, &mut self.data[..]);
+
+        if let Some(notification) = self.data.last() {
+            if !notification.ttl.is_zero() {
+                egui_ctx.request_repaint_after(notification.ttl);
+            }
+        }
+    }
+}
+
+struct NotificationPanel {
+    id: egui::Id,
+}
+
+impl NotificationPanel {
+    fn new() -> Self {
+        Self {
+            id: egui::Id::new("__notifications"),
+        }
+    }
+
+    fn show(
+        &self,
+        egui_ctx: &egui::Context,
+        notifications: &mut Vec<Notification>,
+        is_panel_visible: &mut bool,
+    ) {
+        if !*is_panel_visible {
+            return;
+        }
+
+        for notification in notifications.iter_mut() {
+            notification.ttl = Duration::ZERO;
+        }
+
+        let panel_width = 356.0;
+        let panel_max_height = 320.0;
+
+        let mut to_dismiss = None;
+
+        let notification_list = |ui: &mut egui::Ui| {
+            if notifications.is_empty() {
+                ui.label(
+                    egui::RichText::new("Nothing here!")
+                        .weak()
+                        .color(hex_color!("#636b6f")),
+                );
+
+                return;
+            }
+
+            for (i, notification) in notifications.iter().enumerate().rev() {
+                show_notification(ui, notification, DisplayMode::Panel, || {
+                    to_dismiss = Some(i);
+                });
+            }
+        };
+
+        let mut clear = false;
+
+        egui::Area::new(self.id)
+            .anchor(egui::Align2::RIGHT_TOP, egui::vec2(-8.0, 32.0))
+            .order(egui::Order::Foreground)
+            .interactable(true)
+            .movable(false)
+            .show(egui_ctx, |ui| {
+                egui::Frame::window(ui.style())
```

Comment:
You should be able to use `egui::Window`, which encapsulated an `Area`, `Frame`, and `ScrollArea`. Should just be less code

Reply:
I couldn't figure out how to set `min_scrolled_height` via `egui::Window`